src/messages/variant.rs

```diff
@@ -224,6 +231,8 @@ pub(crate) struct JoinRequest {
     pub section_key: bls::PublicKey,
     /// If the peer is being relocated, contains `RelocatePayload`. Otherwise contains `None`.
     pub relocate_payload: Option<RelocatePayload>,
+    /// Proof of the resouce proofing
+    pub resource_proof: Option<(u64, VecDeque<u8>)>,
```

Comment:
Why is this a `VecDeque` and not a simple `Vec`?

Reply:
Does it make sense for a `JoinRequest` to contain both relocate payload and resource proof? I though that relocated nodes don't have to be proven again, do they? But this type now suggests that such state is valid. To avoid confusion and to make it more robust, I suggest replacing the `relocate_payload` and `resource_proof` fields with a single enum field with three possible variants: "nothing", "relocate_payload" and "resource_proof". Let's say:
```rust
enum JoinDetails {
    None,
    RelocatePayload(RelocatePayload),
    ResourceProof(u64, VecDeque<u8>),
}
```
Alternatively, the resource proof can be a separate message variant.

Reply:
When you got relocated to a new section, that section may need to you to carry out a new round of resource-proofing?
also, the resouce-proofing could be dynamic eventually (i.e. different time/stage may allow nodes to join with different acceptable performance).
So, a re-resour_proofing for relocated node may be necessary ?

Reply:
OK, if you think resource proofing a relocated node is something we might need, then having two separate fields for it makes sense.